src/profile-logic/marker-data.js

```diff
@@ -355,23 +362,51 @@ export function correlateIPCMarkers(threads: Thread[]): IPCMarkerCorrelations {
   // message seqno, and message type. Since the seqno is only unique for each
   // message channel pair, we use the PIDs and message type as a way of
   // identifying which channel pair generated this message.
-  function makeMarkerID(srcPid, destPid, data) {
-    return `${srcPid},${destPid},${data.messageSeqno},${data.messageType}`;
+  function makeMarkerID(thread, data) {
+    let pids;
+    if (data.direction === 'sending') {
+      pids = `${thread.pid},${data.otherPid}`;
+    } else {
+      pids = `${data.otherPid},${thread.pid}`;
+    }
+    return pids + `,${data.messageSeqno},${data.messageType}`;
   }
 
   function formatThreadName(tid) {
     const name = threadNames.get(tid);
     if (name === undefined) {
-      throw new Error('unable to find thread name');
+      return undefined;
     }
     return `${name} (Thread ID: ${tid})`;
   }
 
-  // First, construct a mapping of marker IDs to destination markers for
-  // faster lookup. We also collect the friendly thread names while we have
-  // access to all the threads. It's considerably more difficult to do this
-  // processing later.
-  const dstMarkers: Map<string, Object> = new Map();
+  function phaseToIndex(data): number {
```

Comment:
Please add a short comment explaining why this index is needed. It's a bit magical, and I think this would be a good place to explain it.

Reply:
Done.